apps/api/src/persistence/database/arango-database-for-collection.ts

```diff
@@ -112,6 +112,22 @@ export class ArangoDatabaseForCollection<TEntity extends HasAggregateId> {
             });
     }
 
+    async softDelete(id: string) {
+        await this.#arangoDatabase.softDelete(id, this.collectionID).catch((error) => {
+            throw new InternalError(`failed to soft delete ${this.collectionID}/${id}`, [
+                new InternalError(error?.message || 'unknown arango error'),
```

Comment:
Can't we have an `ArangoError` helper for this? I see this same logic all over the arango implementation.

Reply:
I went ahead and did this one. But I am only leveraging it for this one use case right now. Let's "strangle out" the old pattern as we have the occasion to touch those use cases.